### PR TITLE
fix: check for duplicate schema versions

### PR DESCRIPTION
**Pull request by TheSriram**
- If Duplicates are found, raise a DuplicateSchemaVersionError

### CODE REVIEW

File: cdeploy/migrator.py [hunk 1]
```diff
@@ -29,8 +30,24 @@ def new_migration_filter(f):
...
+
         [self.apply_migration(file_name) for file_name in new_migrations]
 
+    @staticmethod
+    def _find_duplicates(versions):
+        unique = []
+        for version in versions:
+            if version not in unique:
+                unique.append(version)
+        return versions - unique
```
Comment (BryanSD) [inline cdeploy/migrator.py]: This seems to be the source of the test problems:

```python
>>> a = []
>>> b = []
>>> a - b
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for -: 'list' and 'list'
```

**Comment by BryanSD** [comment]
This is a great validation we should be performing. However, I'd like to take it a step further.

As I understand, this only validates migration files beyond the current version. But let's say I'm already at version 6 because I had a file `6_one.cql`. Someone else added a new migration file, `6_two.cql`. With my reading of the code, nothing will error out here since we'll only validate files version 7 and above. However, if anything in files version 7 and above rely on one of the two version 6 files that didn't get applied, we'll have even more schema problems.

I think the best bet is to simply throw our hands up and raise an exception if we find *any* migration files with duplicate versions, even if we've already applied that schema version.

File: cdeploy/migrator.py (hunk 1 above)
Comment (TheSriram) [inline cdeploy/migrator.py]: Yeah Noticed it yesterday evening, looks like that operation can only be performed with sets.

**Comment by TheSriram** [comment]
@BryanSD your scenario is already covered, if the user uses the library as instructed, by trying to do:
```python
    from cdeploy import migrator

    schema_migrator = migrator.Migrator('/path/to/migrations/directory', cassandra_session)
    schema_migrator.run_migrations()
```

Because we check for any duplicates for any version, as evidenced in the new test, I'm going to add.

**Comment by TheSriram** [comment]
hmm, py33 and py34 failed. Let me take a look.

**Comment by TheSriram** [comment]
I'll figure it out.

**Comment by TheSriram** [comment]
@BryanSD, I got your initial point. we are doing the check on new_migrations, but weirdly my test passed in py27. I'll fix that.

**Comment by TheSriram** [comment]
righto, thats better :)

**Comment by BryanSD** [comment]
:+1:

**Pull request merged**